spark/src/main/scala/geotrellis/spark/op/local/spatial/LocalSpatialRasterRDDMethods.scala

```diff
@@ -0,0 +1,24 @@
+package geotrellis.spark.op.local.spatial
+
+import geotrellis.raster.op.local._
+import geotrellis.spark._
+import geotrellis.vector.Geometry
+
+abstract class LocalSpatialRasterRDDMethods[K: SpatialComponent] extends RasterRDDMethods[K] with Serializable {
+
+  /** Masks this raster by the given Geometry. */
+  def mask(geom: Geometry): RasterRDD[K] = {
+    mask(Seq(geom))
+  }
+
+  /** Masks this raster by the given Geometry. */
+  def mask(geoms: Iterable[Geometry]): RasterRDD[K] = {
+    val mapTransform = rasterRDD.metaData.mapTransform
+    rasterRDD.mapPairs { case (k, tile) =>
+      val key = k.spatialComponent
+      val result  = tile.mask(mapTransform(key), geoms)
```

Comment:
Hum, forgot to mention it on Hangouts, but how do you deal with general intersections as it is not defined on all `Geometry` ? Going through `jtsGeom` seems costly.